### PR TITLE
feat: sort out the ordering issues with round tripping models

### PR DESCRIPTION
**Pull request by Midnighter**
Currently, the JSON and YAML writers (inheriting from dict) sort the model attributes. This may cause confusion and is unexpected behaviour. This PR aims to improve that situation and starts out with a series of clear test scenarios.

### CODE REVIEW

File: cobra/test/test_io_order.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+
+@pytest.fixture(scope="module")
+def minimized_core():
+    model = cio.read_sbml_model(join(dirname(__file__), "data",
+                                "textbook.xml.gz"))
+    model.id = "minimized_core"
+    chosen = sample(model.reactions, 10)
+    model.remove_reactions(set(model.reactions).difference(chosen),
+                           remove_orphans=True)
```
Comment (pstjohn) [inline cobra/test/test_io_order.py]: Don't you risk removing all the metabolites if the 10 chosen reactions don't overlap?

Comment (Midnighter) [reply cobra/test/test_io_order.py]: The metabolites of those reactions should remain, right? That seems like enough metabolites to play with.

Comment (pstjohn) [reply cobra/test/test_io_order.py]: Oh I guess you're right, nevermind then, yeah that's probably enough

**Comment by Midnighter** [comment]
Let's see what the tests say but I consider this done. What I finally did:
* Introduce tests
* Make sorting of metabolites, reactions, and genes optional
* Do sort additional attributes such as annotations and notes

**Comment by Midnighter** [comment]
Had to pin the scipy version since the sparse matrices changed quite a bit with the release of 1.0.

**Pull request merged**